src/librustc/hir/lowering.rs

```diff
@@ -1877,18 +1866,18 @@ impl<'a> LoweringContext<'a> {
     // decl: the unlowered (ast) function declaration.
     // fn_def_id: if `Some`, impl Trait arguments are lowered into generic parameters on the
     //      given DefId, otherwise impl Trait is disallowed. Must be `Some` if
-    //      make_ret_async is true.
+    //      make_ret_async is also `Some`.
     // impl_trait_return_allow: determines whether impl Trait can be used in return position.
     //      This guards against trait declarations and implementations where impl Trait is
     //      disallowed.
-    // make_ret_async: if enabled, converts `-> T` into `-> impl Future<Output = T>` in the
+    // make_ret_async: if `Some`, converts `-> T` into `-> impl Future<Output = T>` in the
```

Comment:
Can you leave a comment about what the `NodeId` is used for? At first glance I expected it to be the closure_node_id that is stored in `IsAsync`, and was surprised that it was actually for the `impl Trait`.